html/about.html

```diff
@@ -8,60 +8,130 @@
+                            <tr>
+                                <td>API Identifier</td>
+                                <td id="APIID"></td>
+                                <td></td><td></td>
+                            </tr>
+                        </tbody>
+                    </table>
                 </div>
             </div>
+        </div>
+
+
+        <div class="container pt-5 pb-4 pl-0">
+            <h3>Installed Packages<br></h3>
+        </div>
 
-            
+        <div class="row">
+            <div class="col-xs-12">
+                <div class="table-responsive">
+                    <table class="table">
+                        <thead>
+                            <tr>
+                                <th>Package</th>
+                                <th>Version</th>
+                                <th></th>
+                                <th></th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            {{range .PackageDataRows}}
+                            <tr>
+                                <td>{{index . 0}}</td>
+                                <td class="text-center">{{index . 1}}</td>
+                                <td></td>
+                                <td></td>
+                            </tr>
+                            {{end}}
+                        </tbody>
+                    </table>
+                </div>
+            </div>
         </div>
 
-        <script src="/static/js/jquery-3.3.1.slim.min.js"></script>
-        <script src="/static/js/bootstrap.min.js"></script>
 
-        <script>
-            $(document).ready(function(){
-                document.getElementById("btnBack").href="/advanced";
-                document.getElementById("navDeviceName").href="/advanced";
-            });
-        </script>
+    </div>
+
+    <script src="/static/js/jquery-3.3.1.slim.min.js"></script>
+    <script src="/static/js/bootstrap.min.js"></script>
+
+    <script>
+        window.onload = function() {
+            loadDeviceDetails();
+        };
+
+        $(document).ready(function () {
+            document.getElementById("btnBack").href = "/advanced";
+            document.getElementById("navDeviceName").href = "/advanced";
+        });
+    </script>
+
+    <script>
+        function loadDeviceDetails() {
+            var xmlHttp = new XMLHttpRequest();
+            xmlHttp.open("GET", "/api/device-info", true);
+            xmlHttp.setRequestHeader("Authorization", "Basic "+btoa("admin:feathers"))
+
+            xmlHttp.onload = async function() {
+                if (xmlHttp.status == 200) {
+                    var response = JSON.parse(xmlHttp.response);
+                    $("#APIID").text(response.deviceID)
+                    $("#Group").text(response.groupname)
+                } else {
+                    console.log("error with getting device details");
+                }
+            }
```

Comment:
There's really no reason to dynamically fetch these fields in JavaScript given that they can be retrieved at the time the page template is rendered. Instead of calling `/api/device-info`, use the same functions that `device-info` API handler calls to retrieve these identifiers and substitute them into the template.

Reply:
Do you mean call that from Golang? If so, I don't know how to do that.  I don't really understand how to call the device-info API handler.

Reply:
What I mean is, the code that renders the device info template should call the same things that the `/api/device-info` API handler calls to get those values.

The device-info handler is GetDeviceInfo in api.go. It calls `LoadConfig` and `LoadPrivateConfig` to get the device details. The device info page handler should just call the same functions rather that the more roundabout way of calling the device-info API from JavaScript..

Reply:
Ok I see.